bug: PHP 8.5 compatibility: Infected\array_first() undefined in Infection 0.35.0

| Question    | Answer
| ------------| ---------------
| Infection version | 0.35.0 
| Test Framework version | PHPUnit 13.3.1
| CI | no, local
| PHP version | 8.5.9
| Platform    | macOS (Docker container)
| Github Repo | -

- Used Infection with infection.phar. 
Infection 0.35.0 fails on PHP 8.5 because its bundled symfony/polyfill-php85 does not define Infected\array_first() when running on PHP ≥ 8.5, while Infection still imports and calls that namespaced function.

/bin/infection.phar/src/TestFramework/Coverage/Locator/BaseReportLocator.php:57
/bin/infection.phar/src/TestFramework/Coverage/Locator/BaseReportLocator.php:6
/bin/infection.phar/vendor/symfony/polyfill-php85/bootstrap.php:25
/bin/infection.phar/vendor/symfony/polyfill-php85/bootstrap.php:6

PHP itself has array_first()
php -r 'var_dump(function_exists("array_first"));'
bool(true)

Infection's Infected\array_first() does not exist
php -r 'namespace Infected; require "phar:///app/bin/infection.phar/vendor/symfony/polyfill-php85/bootstrap.php"; var_dump(function_exists("Infected\\array_first"));'
bool(false)


<!-- Please past your phpunit.xml[.dist] if no Github link to the repo provided -->
<details>
 <summary>infection.json5</summary>

 ```json5
{
    "timeout": 20,
    "source": {
        "directories": [
            "src"
        ],
        "excludes": [
            "Kernel.php"
        ]
    },
    "phpUnit": {
        "customPath": "vendor\/bin\/phpunit"
    },
    "tmpDir": "var/cache",
    "logs": {
        "html": "build/infection/infection.html",
        "text": "build/infection/infection.log",
        "summary": "build/infection/summary.log",
        "perMutator": "build/infection/per-mutator.md",
        "gitlab": "build/infection/gitlab-code-quality.json"
    },
    "mutators": {
        "BCMath": true,

        "CloneRemoval": true,

        "PublicVisibility": true,
        "ProtectedVisibility": true,

        "IntegerNegation": true,
        "FloatNegation": true,
        "This": true,
        "FunctionCall": true,
        "NewObject": true,

        "FunctionCallRemoval": true,

        "Foreach_": true,
        "For_": true,
        "Break_": true,
        "Continue_": true,

        "Spaceship": true,

        "Concat": true,
        "SpreadAssignment": true,
        "SpreadRemoval": true,
        "SpreadOneItem": true,

        "Increment": true,
        "Decrement": true,

        "LogicalNot": true,
        "LogicalOr": true,
        "LogicalAnd": true,
        "LogicalLowerAnd": true,
        "LogicalLowerOr": true,
        "Coalesce": true,
        "Equal": true,
        "Identical": true,
        "NotEqual": true,
        "NotIdentical": true,

        "RoundingFamily": true,

        "Throw_": true,
        "Finally_": true,

        "CastArray": true,
        "CastBool": true,
        "CastFloat": true,
        "CastInt": true,
        "CastObject": true,
        "CastString": true,

        "PregMatchMatches": true,

        "MBString": true,

        "UnwrapArrayChangeKeyCase": true,
        "UnwrapArrayChunk": true,
        "UnwrapArrayColumn": true,
        "UnwrapArrayCombine": true,
        "UnwrapArrayDiff": true,
        "UnwrapArrayFilter": true,
        "UnwrapArrayMerge": true,
        "UnwrapArrayReduce": true,
        "UnwrapArrayReplace": true,
        "UnwrapArrayReverse": true,
        "UnwrapArrayUnique": true,
        "UnwrapArrayValues": true,
        "UnwrapStrToLower": true,
        "UnwrapStrToUpper": true,
        "UnwrapTrim": true,
        "UnwrapUcFirst": true,
        "UnwrapUcWords": true
    }
}

 ```
</details>

<details>
 <summary>phpunit.xml</summary>

 ```xml
<?xml version="1.0" encoding="UTF-8"?>
<!-- https://phpunit.readthedocs.io/en/latest/configuration.html -->
<phpunit
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
    colors="true"
    bootstrap="tests/bootstrap.php"
    recordTestRunHistory="false"
    executionOrder="random"
    displayDetailsOnPhpunitDeprecations="true"
    displayDetailsOnTestsThatTriggerDeprecations="true"
    displayDetailsOnTestsThatTriggerErrors="true"
    displayDetailsOnTestsThatTriggerNotices="true"
    displayDetailsOnTestsThatTriggerWarnings="true"
    failOnDeprecation="true"
    failOnNotice="true"
    failOnPhpunitDeprecation="true"
    failOnWarning="true"
>
    <php>
        <ini name="display_errors" value="1"/>
        <ini name="error_reporting" value="-1"/>
        <server name="SHELL_VERBOSITY" value="-1"/>
    </php>
    <testsuites>
        <testsuite name="AllTests">
            <directory>tests</directory>
        </testsuite>
    </testsuites>
    <coverage pathCoverage="true">
        <report>
            <cobertura outputFile="build/coverage/cobertura.xml"/>
            <text outputFile="build/coverage/coverage.txt" showOnlySummary="true"/>
            <html outputDirectory="build/coverage/html"/>
            <xml outputDirectory="build/coverage/xml"/>
        </report>
    </coverage>
    <source>
        <include>
            <directory>src</directory>
        </include>
    </source>
    <logging>
        <junit outputFile="build/coverage/junit.xml"/>
    </logging>
</phpunit>

 ```
</details>

<!-- Remove this section if not needed -->
 ```shell
php bin/infection.phar --threads=8 --coverage=build/ --log-verbosity=default
 ```

<!-- Remove this section if not needed -->
<details>
 <summary>Output with issue</summary>

 ```
 Fatal error: Uncaught Error: Call to undefined function Infected\array_first() in phar:///app/bin/infection.phar/src/TestFramework/Coverage/Locator/BaseReportLocator.php:57
Stack trace:
#0 phar:///app/bin/infection.phar/src/TestFramework/Coverage/Locator/BaseReportLocator.php(28): Infected\Infection\TestFramework\Coverage\Locator\BaseReportLocator->lookup()
#1 phar:///app/bin/infection.phar/src/TestFramework/Coverage/CoverageChecker.php(54): Infected\Infection\TestFramework\Coverage\Locator\BaseReportLocator->locate()
#2 phar:///app/bin/infection.phar/src/Engine.php(68): Infected\Infection\TestFramework\Coverage\CoverageChecker->checkCoverageHasBeenGenerated(''/app/vendor/bi...', 'PHPUnit 13.3.1 ...')
#3 phar:///app/bin/infection.phar/src/Engine.php(44): Infected\Infection\Engine->runInitialTestSuite()
#4 phar:///app/bin/infection.phar/src/Command/RunCommand.php(106): Infected\Infection\Engine->execute()
#5 phar:///app/bin/infection.phar/src/Command/BaseCommand.php(31): Infected\Infection\Command\RunCommand->executeCommand(Object(Infected\Infection\Console\IO))
#6 phar:///app/bin/infection.phar/vendor/symfony/console/Command/Command.php(190): Infected\Infection\Command\BaseCommand->execute(Object(Infected\Symfony\Component\Console\Input\ArgvInput), Object(Infected\Symfony\Component\Console\Output\ConsoleOutput))
#7 phar:///app/bin/infection.phar/vendor/symfony/console/Application.php(714): Infected\Symfony\Component\Console\Command\Command->run(Object(Infected\Symfony\Component\Console\Input\ArgvInput), Object(Infected\Symfony\Component\Console\Output\ConsoleOutput))
#8 phar:///app/bin/infection.phar/vendor/symfony/console/Application.php(254): Infected\Symfony\Component\Console\Application->doRunCommand(Object(Infected\Infection\Command\RunCommand), Object(Infected\Symfony\Component\Console\Input\ArgvInput), Object(Infected\Symfony\Component\Console\Output\ConsoleOutput))
#9 phar:///app/bin/infection.phar/vendor/symfony/console/Application.php(138): Infected\Symfony\Component\Console\Application->doRun(Object(Infected\Symfony\Component\Console\Input\ArgvInput), Object(Infected\Symfony\Component\Console\Output\ConsoleOutput))
#10 phar:///app/bin/infection.phar/src/Console/Application.php(56): Infected\Symfony\Component\Console\Application->run(Object(Infected\Symfony\Component\Console\Input\ArgvInput), Object(Infected\Symfony\Component\Console\Output\ConsoleOutput))
#11 phar:///app/bin/infection.phar/bin/infection(78): Infected\Infection\Console\Application->run()
#12 /app/bin/infection.phar(14): require('phar:///app/bin...')
#13 {main}
  thrown in phar:///app/bin/infection.phar/src/TestFramework/Coverage/Locator/BaseReportLocator.php on line 57

make: *** [infection] Error 255
 ```
</details>

--- a/Makefile
+++ b/Makefile
@@ -15,7 +15,7 @@ help:
 # Variables
 #---------------------------------------------------------------------------
 BOX=./.tools/box
-BOX_URL="https://github.com/humbug/box/releases/download/4.5.1/box.phar"
+BOX_URL="https://github.com/humbug/box/releases/download/4.7.0/box.phar"
 
 PHP_CS_FIXER=./.tools/php-cs-fixer
 PHP_CS_FIXER_URL="https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases/download/v3.92.5/php-cs-fixer.phar"

